### PR TITLE
chore: update CI script to freeze lock file and update yarn.lock

### PR DESCRIPTION
**Pull request by attilah**
*Description of changes:*

After last merge, release task was failing because of differences in ```yarn.lock```, this PR updates a regenerated one and also updates the CI build script to ensure CI build is running with frozen ```yarn.lock``` file.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -13,9 +13,9 @@ jobs:
     <<: *defaults
     steps:
       - checkout
-      - run: yarn cache clean --force
-      - run: yarn
       - run: yarn config set workspaces-experimental true
+      - run: yarn cache clean --force
+      - run: yarn --frozen-lockfile
```
Comment (yuth) [inline .circleci/config.yml]: We don't need this as `yarn production-build` does this

File: .circleci/config.yml [hunk 2]
```diff
@@ -71,7 +71,7 @@ jobs:
             TERM: dumb
       steps:
         - checkout
-        - run: yarn install
+        - run: yarn --frozen-lockfile
```
Comment (yuth) [inline .circleci/config.yml]: probably we can get rid of both checkout and yarn --forzen-lockfile

File: .circleci/config.yml (hunk 1 above)
Comment (attilah) [reply .circleci/config.yml]: Removed

File: .circleci/config.yml [hunk 3]
```diff
@@ -92,14 +92,14 @@ jobs:
               cd ..
               git clone $AUTH_CLONE_URL
               cd aws-amplify-cypress-auth
-              yarn install
+              yarn install --frozen-lockfile
```
Comment (yuth) [inline .circleci/config.yml]: I don't think this is needed. This is a test repo with UI code that we use to run our cypress tests

Comment (attilah) [reply .circleci/config.yml]: Removed install, checkout needs to stay as we discussed.

File: .circleci/config.yml (hunk 2 above)
Comment (attilah) [reply .circleci/config.yml]: Removed install, checkout needs to stay as we discussed.

**Pull request merged**